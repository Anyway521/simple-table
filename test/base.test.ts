import { shallowMount } from "@vue/test-utils";
import { describe, expect, test } from "vitest";
import { ref } from 'vue';
import MyTable from "../src/components/table/index.vue";

describe("mount table", () => {
    const wrapper = shallowMount(MyTable, {
        props: {
            data: ref([
                { 'a': 1 },
                { 'a': 2 }
            ]),
            columns: [
                {
                    title: 'a',
                    key: 'a'
                }
            ],
            defaultHeight: 500
        },
    });
    test('data length', () => {
        expect(wrapper.props().data.value.length).toBe(2)
    });

    test('set height', () => {
        expect(wrapper.props().defaultHeight).toBe(500)
    });

    // 不配置pageAble默认为true
    test('pagable', () => {
        expect(wrapper.props().pageAble).toBe(true)
    });
});
