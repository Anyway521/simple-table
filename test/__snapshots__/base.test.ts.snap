// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"table\\" data-v-6ed10cf1=\\"\\">
  <table class=\\"table_header\\" border=\\"1\\" cellspacing=\\"0\\" cellpadding=\\"0\\" data-v-6ed10cf1=\\"\\">
    <tr class=\\"table_header__row\\" data-v-6ed10cf1=\\"\\">
      <td class=\\"table_header__item\\" title=\\"a\\" data-v-6ed10cf1=\\"\\">a
        <!--v-if-->
        <!--v-if-->
      </td>
    </tr>
  </table>
  <div style=\\"height: 200px;\\" class=\\"table_body__contaniner\\" data-v-6ed10cf1=\\"\\">
    <table class=\\"table_body\\" border=\\"1\\" cellspacing=\\"0\\" cellpadding=\\"0\\" data-v-6ed10cf1=\\"\\">
      <tr class=\\"table_body__row\\" data-v-6ed10cf1=\\"\\"></tr>
      <tr class=\\"table_body__row\\" data-v-6ed10cf1=\\"\\"></tr>
    </table>
    <!--v-if-->
  </div>
</div>
<pagination-stub enable=\\"true\\" total=\\"2\\" pagesize=\\"10\\" data-v-6ed10cf1=\\"\\"></pagination-stub>"
`;

exports[`mount table 1`] = `
"<div class=\\"table\\" data-v-6ed10cf1=\\"\\">
  <table class=\\"table_header\\" border=\\"1\\" cellspacing=\\"0\\" cellpadding=\\"0\\" data-v-6ed10cf1=\\"\\">
    <tr class=\\"table_header__row\\" data-v-6ed10cf1=\\"\\">
      <td class=\\"table_header__item\\" title=\\"a\\" data-v-6ed10cf1=\\"\\">a
        <!--v-if-->
        <!--v-if-->
      </td>
    </tr>
  </table>
  <div style=\\"height: 200px;\\" class=\\"table_body__contaniner\\" data-v-6ed10cf1=\\"\\">
    <table class=\\"table_body\\" border=\\"1\\" cellspacing=\\"0\\" cellpadding=\\"0\\" data-v-6ed10cf1=\\"\\">
      <tr class=\\"table_body__row\\" data-v-6ed10cf1=\\"\\"></tr>
      <tr class=\\"table_body__row\\" data-v-6ed10cf1=\\"\\"></tr>
    </table>
    <!--v-if-->
  </div>
</div>
<pagination-stub enable=\\"true\\" total=\\"2\\" pagesize=\\"10\\" data-v-6ed10cf1=\\"\\"></pagination-stub>"
`;
